src/Compilers/CSharp/Portable/FlowAnalysis/VariablesDeclaredWalker.cs

```diff
@@ -131,12 +131,11 @@ public override void VisitForEachIterationVariables(BoundForEachStatement node)
 
                 if (deconstructionAssignment == null)
                 {
-                    // regular, not deconstructing, foreach declares exactly one iteration variable
-                    _variablesDeclared.Add(node.IterationVariables.Single());
+                    _variablesDeclared.AddAll(node.IterationVariables);
```

Comment:
>_variablesDeclared.AddAll(node.IterationVariables); [](start = 20, length = 51)

Would it be possible to add a targeted compiler test for the affected Region Analysis API?